modules/apps/object/object-api/src/main/java/com/liferay/object/field/business/type/ObjectFieldBusinessType.java

```diff
@@ -154,6 +155,23 @@ public default void validateObjectFieldSettingsDefaultValue(
 			ObjectField objectField,
 			List<ObjectFieldSetting> objectFieldSettings)
 		throws PortalException {
+
+		for (ObjectFieldSetting objectFieldSetting : objectFieldSettings) {
+			if (StringUtil.equals(
+					objectFieldSetting.getName(),
+					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE) &&
+				!(StringUtil.equals(
+					objectFieldSetting.getValue(),
+					ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE) ||
+				  StringUtil.equals(
+					  objectFieldSetting.getValue(),
+					  ObjectFieldSettingConstants.VALUE_EXPRESSION_BUILDER))) {
+
+				throw new ObjectFieldSettingValueException.InvalidValue(
+					objectField.getName(), objectFieldSetting.getName(),
+					objectFieldSetting.getValue());
+			}
+		}
```

Comment:
@murilostodolni @devpedrotavares instead of `List<ObjectFieldSetting> objectFieldSettings`, can we reuse the `Map<String, String> objectFieldSettingsValuesMap`  from `validateObjectFieldSettings` to avoid another `for`?